include/partitioning/parmetis_helper.h

```diff
@@ -73,12 +73,17 @@ class ParmetisHelper
   std::vector<Parmetis::idx_t>  vtxdist;
   std::vector<Parmetis::idx_t>  xadj;
   std::vector<Parmetis::idx_t>  adjncy;
-  std::vector<Parmetis::idx_t>  part;
+  std::vector<dof_id_type>  part;
   std::vector<Parmetis::real_t> tpwgts;
   std::vector<Parmetis::real_t> ubvec;
   std::vector<Parmetis::idx_t>  options;
   std::vector<Parmetis::idx_t>  vwgt;
 
+  // We plan to pass part as a buffer to ParMETIS, so it had better be
+  // using the same type!
+  static_assert(sizeof(Parmetis::idx_t) == sizeof(dof_id_type),
```

Comment:
This static assert is now in two different places, no?

Reply:
Whoops; I'd forgotten I'd insisted on it before the reinterpret_cast before.  I'll leave it in the header next to the documentation of the type choices and I'll remove the old one in the C file.